src/ipa_pc/data_structures.rs

```diff
@@ -518,4 +516,40 @@ impl<F: PrimeField> SuccinctCheckPolynomial<F> {
 
         product
     }
-}
\ No newline at end of file
+}
+
+impl<F: PrimeField> CanonicalSerialize for SuccinctCheckPolynomial<F> {
+
+    #[inline]
+    fn serialize<W: Write>(&self, mut writer: W) -> Result<(), SerializationError> {
+        let len = self.0.len() as u8;
+        CanonicalSerialize::serialize(&len, &mut writer)?;
+        for item in self.0.iter() {
+            // Each field element is, in reality, only 128 bits long
+            let fe128 = item.into_repr().as_ref()[0] as u128 +
```

Comment:
Can you double check that the endianness is what we expect (I don't want that you're taking the 128 bits that are all zero) ?

Reply:
I am not sure I understand what are you asking about. I've checked how data are stored and how conversion functions are implemented in ginger-lib